std/math.d

```diff
@@ -5466,6 +5466,15 @@ real NaN(ulong payload) @trusted pure nothrow @nogc
 }
 
 /**
+ * Returns: true if x isNaN with payload
+ */
+bool isNaNWithPayload(real x, ulong payload) @safe pure nothrow @nogc
+{
+    real other = copysign(NaN(payload), x);
```

Comment:
I think a comment would be nice why this is necessary at all, as it is not obvious. For example "some compilers implement '-x' by a bit operation on the sign bit, but this is inaccurate for NaNs. So better skip the sign bit when comparing the result of passing a NaN through some calculus."